ntfysh_client/NotificationDialog.cs

```diff
@@ -0,0 +1,263 @@
+﻿using System;
+using System.Drawing;
+using System.Runtime.InteropServices;
+using System.Windows.Forms;
+using System.ComponentModel;
+using Microsoft.Win32;
+using System.Diagnostics;
+
+
+namespace ntfysh_client
+{
+    public partial class NotificationDialog : Form
+    {
+        [DllImport("user32.dll", CharSet = CharSet.Auto)]
+        private static extern bool AnimateWindow(IntPtr hWnd, int time, int flags);
+
+        private const int ScreenMargin = 20;
+
+        private int _timeout = 0;
+        private System.Timers.Timer? displayTimeoutTimer = null;
+        private System.Windows.Forms.Timer? updateTimer = null;
+        private Stopwatch? shownStopwatch = null;
+        private ToolTipIcon? _icon;
+        private int _progress_value = 0;
+        private int progress
+        {
+            get { return this._progress_value; }
+            set
+            {
+                this._progress_value = value;
+                this.progressBar1.Value = value;
+            }
+        }
+
+        public bool IsVisible
+        {
```

Comment:
Any reason we're wrapping this? Is it protected usually?

Reply:
Not in the end. Removed. (I had originally though the UI would be `.SetInfo(...); .Show()` or something, but the way it is now has a better expandability and doesn't use this.